AEPLifecycle/Sources/XDM/XDMLifecycleMetricsBuilder.swift

```diff
@@ -0,0 +1,137 @@
+    /// `XDMLifecycleEnvironment` and `XDMLifecycleDevice` info.
+    /// - Returns: App launch event data in dictionary format
+    func buildXDMAppLaunchEventData() -> [String: Any]? {
+        var appLaunchXDMData = XDMMobileLifecycleDetails()
+        appLaunchXDMData.application = xdmApplicationInfoLaunch
+        appLaunchXDMData.device = xdmDeviceInfo
+        appLaunchXDMData.environment = xdmEnvironmentInfo
+        appLaunchXDMData.eventType = LifecycleConstants.XDM.EVENT_TYPE_APP_LAUNCH
+        appLaunchXDMData.timestamp = startDate
+
+        return appLaunchXDMData.asDictionary()
+    }
+
+    /// Builds the data required for the XDM Application Close event, including `XDMLifecycleApplication`
+    /// - Returns: App close event data in dictionary format
+    func buildXDMAppCloseEventData() -> [String: Any]? {
+        var appLaunchXDMData = XDMMobileLifecycleDetails()
+        appLaunchXDMData.application = xdmApplicationInfoClose
+        appLaunchXDMData.eventType = LifecycleConstants.XDM.EVENT_TYPE_APP_CLOSE
+        appLaunchXDMData.timestamp = startDate
+
+        // TODO: MOB-14370 backdate timestamp
+
+        return appLaunchXDMData.asDictionary()
+    }
+
+    /// Adds general application information as well as details related to the type of launch (install, upgrade, regular launch)
+    /// - Parameters:
+    ///   - isInstall: indicates if this is an app install
+    ///   - isUpgrade: indicates if this is an app upgrade
+    /// - Returns: this `XDMLifecycleMetricsBuilder` instance
+    func addAppLaunchData(isInstall: Bool, isUpgrade: Bool) -> XDMLifecycleMetricsBuilder {
+        xdmApplicationInfoLaunch = XDMLifecycleApplication()
+        xdmApplicationInfoLaunch?.isLaunch = true
+
+        if isInstall {
+            xdmApplicationInfoLaunch?.isInstall = true
+            xdmApplicationInfoLaunch?.installDate = startDate
+        }
+
+        if isUpgrade {
+            xdmApplicationInfoLaunch?.isUpgrade = true
+        }
+
+        xdmApplicationInfoLaunch?.name = systemInfoService.getApplicationName()
+        xdmApplicationInfoLaunch?.id = systemInfoService.getApplicationBundleId()
+        xdmApplicationInfoLaunch?.version = systemInfoService.getApplicationVersion()
+
+        return self
+    }
+
+    /// Adds general application information as well as details related to the type of close event
+    /// - Parameters:
+    ///   - previousAppId: application version from the previous session when the close happened
+    ///   - previousSessionInfo: previous session information to be attached to the close event
+    /// - Returns: this `XDMLifecycleMetricsBuilder` instance
+    func addAppCloseData(previousAppId: String?, previousSessionInfo: LifecycleSessionInfo) -> XDMLifecycleMetricsBuilder {
+        xdmApplicationInfoClose = XDMLifecycleApplication()
+
+        xdmApplicationInfoClose?.version = previousAppId
```

Comment:
Same as Android, this should be the previousAppVersion - the appId contains app name + app versions, but we only need the version here. Let's create a task for this and reference it here